### PR TITLE
[MWPW-133840] FEDS check improvements & ES6 support

### PR DESCRIPTION
**Pull request by chadsunvice**
PR relates to ticket [[NALA-Automation] Update FEDS checks (test improvements & ES6 support)](https://jira.corp.adobe.com/browse/MWPW-133840).

Added:
- support for non-US runs to not display the GeoRouting modal (impairing other page actions)
- transitioned to ES6 way of declaring classes/instantiating class objects to match Milo logic
- improved Jarvis checks (extra checks and assertions)

### CODE REVIEW

File: tests/feds/consent.test.js [hunk 1]
```diff
@@ -14,10 +14,10 @@ test.describe('Consent Component test suite', () => {
     await test.step('Navigate to FEDS consent component page', async () => {
       // !Note: Forcing consent to load on GDPR-enforced country.
       // Load OneTrust consent component page:
-      await page.goto(`${baseURL}${features[0].path}?customPrivacyLocation=de`);
+      await page.goto(`${baseURL}${features[0].path}?customPrivacyLocation=de&hideGeorouting=on`);
```
Comment (skumar09) [inline tests/feds/consent.test.js:17]: I think we can define it as a global scoped variable or constant and then use it, instead of changing 
it in every first `test.step()`, of a `test()` 
something like below
`const hideGeoroutingQueryParam = '?hideGeorouting=on'`;
`await page.goto(${baseURL}${features[0].path}${hideGeoroutingQueryParam});`

Also, we can think of defining a property in `.spec.js `and use it in the test.step()
`hideGeo : 'hideGeorouting=on',`
`await page.goto(${baseURL}${features[0].path}${features[0].hideGeo});`

**Review by skumar09 — CHANGES_REQUESTED** [review]

File: tests/feds/jarvis.desktop.test.js [hunk 2]
```diff
@@ -28,12 +28,17 @@ test.describe('Jarvis Component test suite', () => {
...
-      await Jarvis.jarvisButton.click();
+    await test.step('Check Jarvis component functionality', async () => {
+      // Open Jarvis chat assistent (via Jarvis button):
+      await Jarvis.openJarvisFromCta();
+      // Close Jarvis chat assistent (via Jarvis iframe):
+      await Jarvis.closeJarvisFromIframe();
+      // Open Jarvis chat assistent (via header button):
+      await Jarvis.jarvisHeaderButton.click();
       await expect(Jarvis.jarvisContainer).toBeVisible();
+      // Open Jarvis chat assistent (via header button):
```
Comment (amauch-adobe) [inline tests/feds/jarvis.desktop.test.js:39]: Should this say "Close" in the comments instead of "Open" since the check is verifying it isn't visible?

**Review by amauch-adobe — CHANGES_REQUESTED** [review]
Looks really good, just some nitpicks I found, also I agree with @skumar09 that it would be best to use a global/spec/page variable for the URL parameter to hide georouting modal since it is used across many test files.

File: tests/feds/jarvis.desktop.test.js [hunk 3]
```diff
@@ -28,12 +28,17 @@ test.describe('Jarvis Component test suite', () => {
       await Consent.assertOneTrustAcceptState();
     });
 
-    await test.step('Check Jarvis component is initialized', async () => {
-      // Wait for Jarvis component to appear:
-      await Jarvis.jarvisButton.waitFor({ state: 'visible', timeout: 15000 });
-      // Open Jarvis chat assistent:
-      await Jarvis.jarvisButton.click();
+    await test.step('Check Jarvis component functionality', async () => {
+      // Open Jarvis chat assistent (via Jarvis button):
```
[the remark this answers is not in the document]
Comment (chadsunvice) [reply tests/feds/jarvis.desktop.test.js:32]: Haha, yes, I agree @amauch-adobe! The Romanian in me doubled the s but wrote it as in my native language. Old-habits and no spell check in VSCode will do this. 😄

File: tests/feds/jarvis.desktop.test.js [hunk 4]
```diff
@@ -58,12 +63,17 @@ test.describe('Jarvis Component test suite', () => {
...
```
[the remark this answers is not in the document]
Comment (chadsunvice) [reply tests/feds/jarvis.desktop.test.js:67]: Fixed! 👍

File: tests/feds/jarvis.desktop.test.js (hunk 2 above)
Comment (chadsunvice) [reply tests/feds/jarvis.desktop.test.js:39]: Fixed! 👍

File: tests/feds/jarvis.desktop.test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (chadsunvice) [reply tests/feds/jarvis.desktop.test.js:32]: Fixed! 👍

File: tests/feds/consent.test.js (hunk 1 above)
Comment (chadsunvice) [reply tests/feds/consent.test.js:17]: Fixed! Added per-spec file approach since we will have different mixes of browser params which will switch per test type. We already have the consent checks which are a corner-case, so sticking to the second approach. Considered the `feds.config.js` approach, but it was more custom-coding in order to make it work.

File: features/feds/breadcrumbs.spec.js [hunk 5]
```diff
@@ -6,6 +6,7 @@ module.exports = {
       path: [
         '/libs/feds/drafts/qa/breadcrumbs/feds-breadcrumbs-with-base',
       ],
+      browserParams: '?hideGeorouting=on',
```
Comment (skumar09) [inline features/feds/breadcrumbs.spec.js:9]: It is just a perspective, you can take the call 
- from a terminology standpoint, it is referred to as a `query parameter`, you may want to consider using something like `queryParam`, 
- If it is commonly called a `browser parameter` in the applications, then `browserParams` would be easily understood by all

File: tests/feds/header.test.js [hunk 6]
```diff
@@ -2,8 +2,8 @@
 /* eslint-disable no-await-in-loop */
 /* eslint-disable import/extensions */
```
Comment (skumar09) [inline tests/feds/header.test.js:3]: i think` /* eslint-disable import/extensions */` is not required

**Review by skumar09 — APPROVED** [review]
- Minor review comments, approving the PR, 
- I have added a single review comment that applies to all files. Please review and update accordingly

**Review by amauch-adobe — APPROVED** [review]
Approved!

File: features/feds/breadcrumbs.spec.js (hunk 5 above)
Comment (chadsunvice) [reply features/feds/breadcrumbs.spec.js:9]: We usually call them browser params since it happens in the scope of the browser, the full naming being `URL query params`, so we default to `browser params` as a way to shorten it. The naming isn't critical as long as we know that they now reside in the `spec` files.
